docs/proposals/PROPOSAL_A.md

```diff
@@ -0,0 +1,115 @@
+  - **`quotas`** _(bool, OPTIONAL)_ - Allow the filesystem quotas to be changed in the jail.
+  - **`readMsgbuf`** _(bool, OPTIONAL)_ - Jailed users may read the kernel message buffer.
+  - **`socketAf`** _(bool, OPTIONAL)_ - Allow socket types other than IPv4, IPv6 and unix.
+  - **`mlock`** _(bool, OPTIONAL)_ - Allow locking and unlocking of physical pages.
+  - **`nfsd`** _(bool, OPTIONAL)_ - Allow the jail to act as an NFS server.
+  - **`reservedPorts`** _(bool, OPTIONAL)_ - Allow the jail to bind to ports lower than 1024.
+  - **`suser`** _(bool, OPTIONAL)_ - The value of the jail's security.bsd.suser_enabled sysctl. The super-user will be disabled automatically if its parent system has it disabled.  The super-user is enabled by default.
+  
+### Mapping from jail(8) config file
+
+This table shows how to map settings from a typical jail(8) config file to the proposed JSON format.
+
+| Jail parameter | JSON equivalent      |
+| -------------- | -------------------- |
+| jid            | -                    |
+| name           | see below            |
+| path           | root.path            |
+| ip4.addr       | freebsd.jail.ip4Addr |
+| ip4.saddrsel   | -                    |
+| ip4            | freebsd.jail.ip4     |
+| ip6.addr       | freebsd.jail.ip6Addr |
+| ip6.saddrsel   | -                    |
+| ip6            | freebsd.jail.ip6     |
+| vnet           | freebsd.jail.vent    |
+| host.hostname  | hostname             |
+| host           | freebsd.jail.host    |
+| sysvmsg        | freebsd.jail.sysvmsg |
+| sysvsem        | freebsd.jail.sysvsem |
+| sysvshm        | freebsd.jail.sysvshm |
+| securelevel    | -                    |
+| devfs_ruleset  | see below            |
+| children.max   | see below            |
+| enforce_statfs | freebsd.jail.enforceStatfs |
+| persist        | -                    |
+| parent         | freebsd.jail.parent  |
+| osrelease      | -                    |
+| osreldate      | -                    |
+| allow.set_hostname | freebsd.jail.allow.setHostname |
+| allow.sysvipc  | freebsd.jail.allow.sysvipc |
+| allow.raw_sockers  | freebsd.jail.allow.rawSockets |
+| allow.chflags  | freebsd.jail.allow.chflags |
+| allow.mount    | freebsd.jail.allow.mount |
+| allow.quotas    | freebsd.jail.allow.quotas |
+| allow.read_msgbuf | freebsd.jail.allow.readMsgbuf |
+| allow.socket_af | freebsd.jail.allow.socketAf |
+| allow.mlock    | freebsd.jail.allow.mlock |
+| allow.nfsd     | freebsd.jail.allow.nfsd |
+| allow.reserved_ports | freebsd.jail.allow.reservedPorts |
+| allow.unprivileged_proc_debug | - |
+| allow.suser    | freebsd.jail.allow.suser |
+| allow.mount.*  | see below            |
+| securelevel    | -                    |
+
+The jail name is set to the create command's `container-id` argument.
+
+The `devfs_ruleset` parameter is only required for jails which create new `devfs` mounts - typically OCI runtimes will mount `devfs` on the host.
```

Comment:
How would the ruleset itself be specified?  Can that be defined through the config?

Reply:
For the container jail's devfs, which is mounted by the host, the ruleset is set as a mount option and defaults to ruleset 4. I'm not sure how useful the devfs_ruleset parameter is for container jails, possibly for nesting? I wasn't going to try to map this one to the json config, at least until we have a use-case which needs it.

Reply:
I think just having this as numbers requires that whatever higher-level tool is creating the OCI bundle is aware of the existing devfs rulesets on the host.  Even if we don't want to provide a mechanism to author rulesets from within a bundle, we should call this out as a requirement for the higher-level tool.

Reply:
The main use-case for this is probably nesting containers - in that case, perhaps we could use the container's ruleset number as a default for devfs_ruleset?